From ef638a84405c9f6556a9d7c257ccbba74efd228e Mon Sep 17 00:00:00 2001
From: Ansuel Smith <ansuelsmth@gmail.com>
Date: Sat, 14 May 2022 20:15:10 +0200
Subject: [PATCH 10/12] frontends: drop udp_get_timeouts and use standard
 upstream api

Drop udp_get_timeouts and use nf_udp_pernet and ->timeouts
instead or relying on a downstream api not present upstream.
---
 frontends/nss/ecm_nss_ipv4.c | 3 ++-
 frontends/nss/ecm_nss_ipv6.c | 3 ++-
 frontends/sfe/ecm_sfe_ipv4.c | 3 ++-
 frontends/sfe/ecm_sfe_ipv6.c | 3 ++-
 4 files changed, 8 insertions(+), 4 deletions(-)

--- a/frontends/nss/ecm_nss_ipv4.c
+++ b/frontends/nss/ecm_nss_ipv4.c
@@ -610,7 +610,8 @@ sync_conntrack:
 #else
 				timeouts = nf_ct_timeout_lookup(ct);
 				if (!timeouts) {
-					timeouts = udp_get_timeouts(nf_ct_net(ct));
+					struct nf_udp_net *un = nf_udp_pernet(nf_ct_net(ct));
+					timeouts = un->timeouts;
 				}

 				spin_lock_bh(&ct->lock);
--- a/frontends/nss/ecm_nss_ipv6.c
+++ b/frontends/nss/ecm_nss_ipv6.c
@@ -587,7 +587,8 @@ sync_conntrack:
 #else
 				timeouts = nf_ct_timeout_lookup(ct);
 				if (!timeouts) {
-					timeouts = udp_get_timeouts(nf_ct_net(ct));
+					struct nf_udp_net *un = nf_udp_pernet(nf_ct_net(ct));
+					timeouts = un->timeouts;
 				}

 				spin_lock_bh(&ct->lock);
--- a/frontends/sfe/ecm_sfe_ipv4.c
+++ b/frontends/sfe/ecm_sfe_ipv4.c
@@ -562,7 +562,8 @@ sync_conntrack:
 #else
 				timeouts = nf_ct_timeout_lookup(ct);
 				if (!timeouts) {
-					timeouts = udp_get_timeouts(nf_ct_net(ct));
+					struct nf_udp_net *un = nf_udp_pernet(nf_ct_net(ct));
+					timeouts = un->timeouts;
 				}

 				spin_lock_bh(&ct->lock);
--- a/frontends/sfe/ecm_sfe_ipv6.c
+++ b/frontends/sfe/ecm_sfe_ipv6.c
@@ -556,7 +556,8 @@ sync_conntrack:
 #else
 				timeouts = nf_ct_timeout_lookup(ct);
 				if (!timeouts) {
-					timeouts = udp_get_timeouts(nf_ct_net(ct));
+					struct nf_udp_net *un = nf_udp_pernet(nf_ct_net(ct));
+					timeouts = un->timeouts;
 				}

 				spin_lock_bh(&ct->lock);
